### PR TITLE
Compile/runtime warnings support

### PR DESCRIPTION
**Pull request by pfalcon**

### CODE REVIEW

**Comment by pfalcon** [comment]
I wanted to have warnings support for a long time, but there were not enough "mainline-ready" usecases for it. Finally I was able to find whole 2 roughly (!) fitting cases for it in different parts of code, but as commits show, had to disable one already, and another really targeted for disabling too (because as the original message says, it's notice, not a warning).

Nonetheless, I'd like to have (more or less) structured support for them, for development/future/experimental work. For example, one usecase I always had in mind is to add "static check" option, which would issue warnings if a var assigned different types during rumtime.

File: py/warning.c [hunk 1]
```diff
@@ -0,0 +1,29 @@
+#include <stdint.h>
```
Comment (dpgeorge) [inline py/warning.c]: License header?

**Comment by dpgeorge** [comment]
This is fine.

We could modify the compiler so that it doesn't emit delete opcode for exceptions compiled using native codgen, but that would cost bytes.

**Comment by pfalcon** [comment]
Added header, merging then. I agree that if we want to spend more bytes, there're more interesting ways - for example, making this automatically-to-be-deleted variable always present, so raise w/o arguments can be implemented ;-) (which is another small, but hard problem I looked into (btw, it was not exactly seamless for interpreter too - [mp_exc_stack_t::prev_exc](https://github.com/micropython/micropython/blob/master/py/bc.h#L34 exists only to support raise w/o args))).

**Pull request closed without merging**